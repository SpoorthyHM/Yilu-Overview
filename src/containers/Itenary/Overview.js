import React, {Component} from 'react';
import Layout from '../../hoc/Layout/Layout';
import axios from '../../axios-itenary';
import Itinerary from '../../components/Itinerary/Itinerary';

class Overview extends Component {

    state = {
        itenaryDetails: {}
    }

    initItenaryDetails = () => {
        
        axios.get('/itenary')
        .then(response => {
            this.setState({itenaryDetails: response.data});
        })
        .catch(error => {
            console.log('Error');
        });
    }

    componentDidMount() {
        this.initItenaryDetails();
    }
    
    render(){
        var itenary = {};
        if(this.state.itenaryDetails) {
            itenary = this.state.itenaryDetails.itenaryDetails;
        }
        
        return (
            <Layout>
                <div className="card">
                    <div className="card-header bg-primary text-white">Overview</div>
                    <div className="card-body bg-light text-dark">
                        Berlin - Munchen
                                <Itinerary details={itenary}/>
                    </div> 
                </div>                                             
            </Layout>
        );
    }
}

export default Overview;