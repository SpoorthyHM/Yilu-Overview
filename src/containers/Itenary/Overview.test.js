import React from 'react';
import { shallow} from 'enzyme';
import Overview from './Overview';

describe('<Overview />', () => {
    it('should render correctly', () => {
        const component = shallow(<Overview />);
        expect(component).toMatchSnapshot();
    });
  
})