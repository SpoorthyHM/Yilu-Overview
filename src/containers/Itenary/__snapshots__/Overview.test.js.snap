// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overview /> should render correctly 1`] = `ShallowWrapper {}`;
